$OpenBSD: patch-Source_WebCore_contentextensions_NFAToDFA_cpp,v 1.1 2019/03/25 14:03:49 cwen Exp $

unbreak the build where char is unsigned by default (arm, ppc)

Index: Source/WebCore/contentextensions/NFAToDFA.cpp
--- Source/WebCore/contentextensions/NFAToDFA.cpp.orig
+++ Source/WebCore/contentextensions/NFAToDFA.cpp
@@ -40,10 +40,10 @@
 namespace WebCore {
 
 namespace ContentExtensions {
-
-typedef MutableRange<char, NFANodeIndexSet> NFANodeRange;
-typedef MutableRangeList<char, NFANodeIndexSet> NFANodeRangeList;
-typedef MutableRangeList<char, NFANodeIndexSet, 128> PreallocatedNFANodeRangeList;
+//error: non-constant-expression cannot be narrowed from type 'char' to 'signed char'
+typedef MutableRange<signed char, NFANodeIndexSet> NFANodeRange;
+typedef MutableRangeList<signed char, NFANodeIndexSet> NFANodeRangeList;
+typedef MutableRangeList<signed char, NFANodeIndexSet, 128> PreallocatedNFANodeRangeList;
 typedef Vector<uint32_t, 0, ContentExtensionsOverflowHandler> UniqueNodeList;
 typedef Vector<UniqueNodeList, 0, ContentExtensionsOverflowHandler> NFANodeClosures;
 
